### PR TITLE
issue-29621: Excluding Hidden content types from the Widgets displayed

### LINKED ISSUE #29621: Be able to hide "Forms" from the Content tab in Edit mode.
### Parent Issue

_No response_

### User Story

As a page designer, I want to be able to hide Forms from the content tab in the Page edit mode, so I can only see the exact content types I want to go into containers. 

I have included the screenshot provided by the customer to show exactly where. 

### Acceptance Criteria

Be able to hide the "Forms" content type 

### Proposed Objective

User Experience

### Proposed Priority

Priority 4 - Trivial

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/117725

Forms component being shown

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by Neehakethi**
This pull request introduces several changes to the `DotPaletteStore` and related test files to enhance the handling of content types and widgets, including the filtering of hidden content types. The most important changes include adding the `DotPropertiesService` for configuration management and refactoring the content type loading logic.

Enhancements to `DotPaletteStore`:

* Added `DotPropertiesService` to manage configuration settings and retrieve hidden content types (`CONTENT_PALETTE_HIDDEN_CONTENT_TYPES`). (`[[1]](diffhunk://#diff-8262303c46e34e159efdc68dec99a47ddf453302794be94483b1abee94c13556R9)`, `[[2]](diffhunk://#diff-5e9e57d14e6a9185a0988e2062be4028f862924c1b068ac9eddab73da058e94fR13-R20)`, `[[3]](diffhunk://#diff-5e9e57d14e6a9185a0988e2062be4028f862924c1b068ac9eddab73da058e94fL166-R169)`)
* Refactored the `getContenttypesData` method to use `forkJoin` with multiple observables, including hidden content types, and improved the filtering logic to exclude hidden content types. (`[core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.tsL252-L282](diffhunk://#diff-5e9e57d14e6a9185a0988e2062be4028f862924c1b068ac9eddab73da058e94fL252-L282)`)

Updates to tests:

* Updated test cases to include `DotPropertiesService` and verify the filtering of hidden content types in `dot-palette.store.spec.ts` and `edit-ema-palette.store.spec.ts`. (`[[1]](diffhunk://#diff-8262303c46e34e159efdc68dec99a47ddf453302794be94483b1abee94c13556R114-R136)`, `[[2]](diffhunk://#diff-8262303c46e34e159efdc68dec99a47ddf453302794be94483b1abee94c13556R147)`, `[[3]](diffhunk://#diff-8262303c46e34e159efdc68dec99a47ddf453302794be94483b1abee94c13556L185-R233)`, `[[4]](diffhunk://#diff-1440c725f71c6ba67d7a4847e983e63b1e7a3a4fe17042aafe9553a89ad19f22L46-R82)`, `[[5]](diffhunk://#diff-1440c725f71c6ba67d7a4847e983e63b1e7a3a4fe17042aafe9553a89ad19f22R119-R131)`, `[[6]](diffhunk://#diff-1440c725f71c6ba67d7a4847e983e63b1e7a3a4fe17042aafe9553a89ad19f22R141-R160)`)

Refactoring:

* Replaced inline sorting and filtering logic with pre-defined constants and helper methods to improve code readability and maintainability. (`[[1]](diffhunk://#diff-8262303c46e34e159efdc68dec99a47ddf453302794be94483b1abee94c13556R114-R136)`, `[[2]](diffhunk://#diff-8262303c46e34e159efdc68dec99a47ddf453302794be94483b1abee94c13556L185-R233)`, `[[3]](diffhunk://#diff-e60190c318aa61ddbc495b1c20398bb42e7baa8429b8241aa6964267f3ddd445L235-L259)`)

These changes improve the overall functionality and maintainability of the `DotPaletteStore` by incorporating configuration management and enhancing the content type filtering logic.

### Video

https://github.com/user-attachments/assets/a376dd9d-861f-4250-a67d-1dd0dd5474da

This PR fixes: #29621

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/edit-ema-palette/store/edit-ema-palette.store.ts [hunk 1]
```diff
@@ -237,10 +237,11 @@ export class DotPaletteStore extends ComponentStore<DotPaletteState> {
                 filter,
                 page: 40,
                 type: 'WIDGET'
-            })
+            }),
+            this.dotConfigurationService.getKeyAsList('CONTENT_PALETTE_HIDDEN_CONTENT_TYPES')
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/edit-ema-palette/store/edit-ema-palette.store.ts:241]: Is this a new variable or do we already had that?

**Review by fmontes — APPROVED** [review]
Approved, but need docs and needs unit tests.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/edit-ema-palette/store/edit-ema-palette.store.ts (hunk 1 above)
Comment (erickgonzalez) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/edit-ema-palette/store/edit-ema-palette.store.ts:241]: We already had it for a while

**Comment by zJaaal** [comment]
The code LGTM but we need a unit test to merge it

**Review by rjvelazco — COMMENTED** [review]
@erickgonzalez is this still needed it?

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/edit-ema-palette/store/edit-ema-palette.store.ts [hunk 2]
```diff
@@ -249,9 +250,13 @@ export class DotPaletteStore extends ComponentStore<DotPaletteState> {
...
                     (item) => item.baseType !== 'WIDGET'
                 );
+                // Filter widgets based to exclude hidden content types
+                const relevantWidgets = widgets.filter(
+                    (widget) => !hiddenContentTypes.includes(widget.variable)
+                );
 
                 // Merge both array and order them by name
-                const contentTypes = [...contentLets, ...widgets]
+                const contentTypes = [...contentLets, ...relevantWidgets]
```
[the remark this answers is not in the document]
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/edit-ema-palette/store/edit-ema-palette.store.ts]: Done!

**Review by zJaaal — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Pull request merged**